### ISSUE TITLE
context fails when comparing units

### ISSUE THREAD

**Issue by mmarras** [hgrecco/pint#1748]
I'm trying to use context to compare two units but it fails for pint 0.20.1

Let's define the context on earth
```python
import pint
def ureg_f() -> pint.UnitRegistry:
    """Provide ureg class from pint but adds some special units."""
    ureg = pint.UnitRegistry()
    # activate matplotlib compatibility layer
    ureg.setup_matplotlib()
    # define earth context so force can be converted to mass assuming standard gravity acceleration
    c_earth = pint.Context("earth")
    c_earth.add_transformation("[force]", "[mass]", lambda ureg, x: x / ureg.g_n)
    c_earth.add_transformation("[mass]", "[force]", lambda ureg, x: x * ureg.g_n)
    ureg.add_context(c_earth)

ureg = uref_f()
```

so then this works as expected:
```python
with ureg.context('earth'):
    ureg.Quantity(1,'kg').to('kN')
```

however: 
```python
with ureg.context('earth'):
    ureg.Quantity(1,'kN')<= ureg.Quantity(10,'kg')
```
throws 

> DimensionalityError: Cannot convert from 'kilonewton' ([length] * [mass] / [time] ** 2) to 'kilogram' ([mass])

I would expect it to convert to compareable units because `ureg.Quantity(1,'kN')<= ureg.Quantity(10,'force_pound')` something like this works too.

**Comment by MichaelTiemannOSC** [comment]
I've tripped into this as well.  What I've learned is that pint won't cross the dimensionality divide without a strong request (like a .to() call).  That's just how it is.

**Comment by hgrecco** [comment]
Indeed. Contexts are only implemented for `to` right now. We could think about adding this for other operators if there is a reall need.

**Comment by mmarras** [comment]
My usecase is that I have a class which takes quantities as initial values which come from a user interface where the user can select different units that are „compatible“ with those parameters. 

I can of course coerce all input to the expected units first with a .to, but much more elegant would be to be able to just wrap that initiation call into a „with context“ and be done.
